### PR TITLE
few fixes to endpointslice controller's repair code

### PR DESCRIPTION
**Pull request by girishmg**
This PR has two commits:

First commit:
```
capture the fact that OVN GR LB is empty as an error object

we need this in several places and we were trying to compare the error
string, and we should avoid it.
```

Second commit:
```
- since we run the repair only once at the beginning, we should clean
  up as much as possible and logging error where we can't and still
  proceed
- some error messages were printing the entire LB UUID list on failure,
  we only need to print only one LB
```

### CODE REVIEW

File: go-controller/pkg/ovn/gateway/gateway.go [hunk 1]
```diff
@@ -15,6 +15,12 @@ const (
 	OvnGatewayLoadBalancerIds = "lb_gateway_router"
 )
 
+var (
+	// It is perfectly noraml to have OVN GW routers to not to have LB rules. This happens
```
Comment (aojea) [inline go-controller/pkg/ovn/gateway/gateway.go:19]: typo noraml/normal

**Comment by aojea** [comment]
interesting debate :)

why should we continue if we don't have the ovn "infrastructure" ready?

I see that failing fast gives better information to the cluster administrator, they can say "hey my ovn master is pod is not working", or "is crashlooping", and it has a point to start to investigate.
If we continue only logging, you just have all the pods working and people is going to open a ticket with the "I can't access the services in the cluster but all pods are running" ... my experience is that nobody is going to see the logs of the repair loop saying that there was an error ...

File: go-controller/pkg/ovn/gateway/gateway.go (hunk 1 above)
Comment (girishmg) [reply go-controller/pkg/ovn/gateway/gateway.go]: fixed

**Comment by girishmg** [comment]
The discussion/debate should be in the context of Repair :-). The definition of repair is to remove any stale data from OVN DB so that we are not carrying unnecessary stuff in the DBs. 

So, keeping that in perspective. I think we should just log errors and proceed to clean as much as we can.

Prior to this PR, we were doing the same. That is, we just logged error that we couldn't repair the service.
```
	if err := c.repair.RunOnce(); err != nil {
		klog.Errorf("Error repairing services: %v")
	}
```
This PR, makes `repair` a best effort.


For as to critical paths -- pod creation, node addition, and so on.. we already deploy `retries` and we not just log. So, I don't see any issues there.

**Comment by aojea** [comment]
> ```
> 	if err := c.repair.RunOnce(); err != nil {
> 		klog.Errorf("Error repairing services: %v")
> 	}
> ```
> 
> 
> This PR, makes `repair` a best effort.

lol,
disregard my comment, don't know why I was thinking it failed, :)

**Pull request merged**